examples/vision/image_classification_efficientnet_fine_tuning.py

```diff
@@ -0,0 +1,423 @@
+"""
+Title: Image classification using EfficientNet and fine-tuning
+Author: Yixing Fu
+Date created: 2020/06/30
+Last modified: 2020/07/06
+Description: Use EfficientNet with weights pre-trained on imagenet for CIFAR-100 classification.
+"""
+"""
+
+## What is EfficientNet
+EfficientNet, first introduced in https://arxiv.org/abs/1905.11946 is among the most
+efficient models (i.e. requiring least FLOPS for inference) that reaches SOTA in both
+imagenet and common image classification transfer learning tasks.
+
+The smallest base model is similar to MnasNet (https://arxiv.org/abs/1807.11626), which
+reached near-SOTA with a significantly smaller model. By introducing a heuristic way to
+scale the model, EfficientNet provides a family of models (B0 to B7) that represents a
+good combination of efficiency and accuracy on a variety of scales. Such a scaling
+heuristics (compound-scaling, details see https://arxiv.org/abs/1905.11946) allows the
+efficiency-oriented base model (B0) to surpass models at every scale, while avoiding
+extensive grid-search of hyperparameters.
+
+A summary of the latest updates on the model is available at
+https://github.com/tensorflow/tpu/tree/master/models/official/efficientnet, where various
+augmentation schemes and semi-supervised learning approaches are applied to further
+improve the imagenet performance of the models. These extensions of the model can be used
+by updating weights without changing model architecture.
+
+## Compound scaling
+
+The EfficientNet models are approximately created using compound scaling. Starting from
+the base model B0, as model size scales from B0 to B7, the extra computational resource
+is proportioned into width, depth and resolution of the model by requiring each of the
+three dimensions to grow at the same power of a set of fixed ratios.
+
+However, it must be noted that the ratios are not taken accurately. A few points need to
+be taken into account:
+Resolution. Resolutions not divisible by 8, 16, etc. cause zero-padding near boundaries
```

Comment:
Probably add a line break or 1/2/3 here, to make individual points more readable.

Reply:
added.